### DISCUSSION TITLE
Random seeding

### DISCUSSION THREAD

**Discussion by jbarr444** [open-atmos/PySDM#1338] [General]
Hi,

I am trying to run the Shipway_and_Hill_2012 example multiple times to examine random behavior. I have basically just put cell 3 in the notebook example inside a for loop. This produces a bunch of identical simulations, but when rerun, produces a (slightly) different set of identical simulations. What's going on here? Can it be made so that each simulation is independent of the others?

**Comment by slayoo** [comment]
@jbarr444 thank you for trying PySDM out and for reporting feedback!
PySDM sets the default random seed on package import (https://github.com/open-atmos/PySDM/blob/main/PySDM/physics/constants.py#L49-L53). So, unless a different random seed is provided manually (through `Formulae` class ctor), rerunning the simulations in the same Python/Jupyter session will produce exactly the same results. However, restarting a notebook will reimport PySDM, and hence set a new random seed.

In fact, we already have an issue reported suggesting to change this behaviour: https://github.com/open-atmos/PySDM/issues/1180
